modopt/opt/proximity.py

```
@@ -688,3 +689,341 @@ def _cost_method(self, *args, **kwargs):
+            cnt += 1
+
+            if prev_midpoint == midpoint:
+
+                # Particular case
+                sum_0 = self._compute_theta(data_abs, alpha[first_idx],
+                                            extra_factor).sum()
+                sum_1 = self._compute_theta(data_abs, alpha[last_idx],
+                                            extra_factor).sum()
+
+                if (np.abs(sum_0 - self.k_value) <= 1e-4):
+                    found = True
+                    midpoint = first_idx
+
+                if (np.abs(sum_1 - self.k_value) <= 1e-4):
+                    found = True
+                    midpoint = last_idx - 1
+                    # -1 because output is index such that
+                    # sum(theta(alpha[index])) <= k
+
+                if (first_idx - last_idx == 2) or (first_idx - last_idx == 1):
+                    sum_0 = self._compute_theta(data_abs, alpha[first_idx],
+                                                extra_factor).sum()
+                    sum_1 = self._compute_theta(data_abs, alpha[last_idx],
+                                                extra_factor).sum()
+                    if (sum_0 <= self.k_value) or (sum_1 >= self.k_value):
+                        found = True
+
+            sum_0 = self._compute_theta(data_abs, alpha[midpoint],
+                                        extra_factor).sum()
+            sum_1 = self._compute_theta(data_abs, alpha[midpoint + 1],
+                                        extra_factor).sum()
+
+            if (sum_0 <= self.k_value) & (sum_1 >= self.k_value):
+                found = True
+
+            elif sum_1 < self.k_value:
+                first_idx = midpoint
+
+            elif sum_0 > self.k_value:
+                last_idx = midpoint
+
+            prev_midpoint = midpoint
+
+        if found:
+            return midpoint, alpha[midpoint], alpha[midpoint + 1], sum_0,\
+                   sum_1
+        else:
+            return -1
```

Probably need to raise an error if it has not been found, rather than setting the output to -1, which will cause unhandled error

---

Done